src/classes/RD2_OpportunityService.cls

```diff
@@ -237,23 +248,63 @@ public with sharing class RD2_OpportunityService {
                 currentDate, matcher.getNumberOfOpportunities(), rd.RecurringDonationSchedules__r
             ));
 
+            // Track unmatched matcher records so we can connect Opps and RDs from RDs with altered schedules
+            Boolean openOpportunityMatched = false;
+            Opportunity earliestUnmatchedOpportunity;
+            RD2_ScheduleService.Installment earliestUnmatchedInstallment;
+
             for (RD2_OpportunityMatcher.Record record : matcher.getRecords()) {
+
+                if (record.hasInstallment() && record.isNew()) {
```

Comment:
There's a lot in this for loop body, plus it's a nested loop. Do you think it's at the point were it should be abstracted into a new method - maybe something like `updateOpportunityInstallmentRecord`?

Reply:
Yes, I agree this has become cumbersome.